### PR TITLE
Font AutoScalingEnabled

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###

Implements #1683

For Android/WinUI the default behavior here doesn't change. 

This now means that fonts on iOS will auto scale based on the dynamic font sizing settings on the device

image

### Additions made ###
- Adds Font.AutoScalingEnabled
- Adds Button.FontAutoScalingEnabled
- Adds DatePicker.FontAutoScalingEnabled
- Adds Editor.FontAutoScalingEnabled
- Adds Entry.FontAutoScalingEnabled
- Adds FontElement.FontAutoScalingEnabled
- Adds Label.FontAutoScalingEnabled
- Adds Picker.FontAutoScalingEnabled
- Adds RadioButton.FontAutoScalingEnabled
- Adds SearchBar.FontAutoScalingEnabled
- Adds Span.FontAutoScalingEnabled
- Adds TimePicker.FontAutoScalingEnabled
- Adds FontImageSource.FontAutoScalingEnabled

- (Android) Adds 
```C#
	public struct FontSize
	{
		public FontSize(float value, ComplexUnitType unit)
		{
			Value = value;
			Unit = unit;
		}

		public float Value { get; set; }
		public ComplexUnitType Unit { get; set; }
	}
```
- (Android) Changes `float IFontManager.GetSize` => `FontSize IFontManager.GetSize`

### Removed ###
- Button.Font (to make Button consistent with all the other controls)
- (iOS) IFontManager.GetSize  Nothing outside of FontManager is currently using this so I just removed it for now

### CODE REVIEW

File: src/Controls/src/Core/FontElement.cs [hunk 1]
```diff
@@ -108,14 +116,36 @@ static void OnFontAttributesChanged(BindableObject bindable, object oldValue, ob
...
+			var fontAttributes = (FontAttributes)bindable.GetValue(FontAttributesProperty);
+			var fontFamily = (string)bindable.GetValue(FontFamilyProperty);
+			var enableAutoScaling = (bool)newValue;
+
+			if (fontFamily != null)
+				bindable.SetValue(FontProperty, Font.OfSize(fontFamily, fontSize, enableScaling: enableAutoScaling).WithAttributes(fontAttributes));
+			else
+				bindable.SetValue(FontProperty, Font.SystemFontOfSize(fontSize, enableScaling: enableAutoScaling).WithAttributes(fontAttributes));
+
+			SetCancelEvents(bindable, false);
```
Comment (mattleibow) [inline src/Controls/src/Core/FontElement.cs:147]: Is it not possible to refactor all these methods into a single one and call it? I worry we might make a change in one place and forget another, also will delete like 50 lines of code.

File: src/Controls/src/Core/Span.cs [hunk 2]
```diff
@@ -76,6 +76,8 @@ public string Text
 
 		public static readonly BindableProperty FontAttributesProperty = FontElement.FontAttributesProperty;
 
+		public static readonly BindableProperty FontScalingEnabledProperty = FontElement.FontScalingEnableProperty;
```
Comment (mattleibow) [inline src/Controls/src/Core/Span.cs:79]: Is this possible? To have some text and only scale a span? On iOS this property is for the label only and might not be possible.

File: src/Controls/src/Core/FontImageSource.cs [hunk 3]
```diff
@@ -42,5 +42,15 @@ public double Size
 			get => (double)GetValue(SizeProperty);
 			set => SetValue(SizeProperty, value);
 		}
+
+		public static readonly BindableProperty ScalingEnableProperty =
+			BindableProperty.Create("ScalingEnable", typeof(bool), typeof(FontImageSource), false,
+				propertyChanged: (b, o, n) => ((FontImageSource)b).OnSourceChanged());
+
+		public bool ScalingEnable
```
Comment (mattleibow) [inline src/Controls/src/Core/FontImageSource.cs:50]: Not sure if this is totally supported since the image will have to be re-generated when the accessibility settings change. I think we have support for when the density/resolution changes, so this is not crazy.

File: src/Core/src/Fonts/FontSize.Android.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Android.Util;
+
+namespace Microsoft.Maui
+{
+	public struct FontSize
```
Comment (mattleibow) [inline src/Core/src/Fonts/FontSize.Android.cs:8]: Happy thoughts.

File: src/Core/src/Primitives/Font.cs [hunk 5]
```diff
@@ -23,6 +23,13 @@ public FontWeight Weight
 			private set => _weight = value;
 		}
 
+		bool _disableScaling;
+		public bool EnableScaling
```
Comment (mattleibow) [inline src/Core/src/Primitives/Font.cs:27]: `EnableScaling` or `EnableAutoScaling`... Not sure.

File: src/Controls/src/Core/FontImageSource.cs (hunk 3 above)
Comment (mattleibow) [reply src/Controls/src/Core/FontImageSource.cs:50]: Also the name is different to the other properties.

File: src/Controls/src/Core/Button.cs [hunk 6]
```diff
@@ -116,12 +116,6 @@ public object CommandParameter
 			set { SetValue(CommandParameterProperty, value); }
 		}
 
-		public Font Font
```
Comment (PureWeen) [inline src/Controls/src/Core/Button.cs:119]: this was deleted on all of the other controls (span,label, etc..) so I'm assuming this one was left here by mistake

File: src/Controls/src/Core/FontElement.cs (hunk 1 above)
Comment (PureWeen) [reply src/Controls/src/Core/FontElement.cs:147]: Probably, I just went for the quick version right now and I'll look at refactoring this tomorrow

File: src/Controls/src/Core/FontImageSource.cs (hunk 3 above)
Comment (PureWeen) [reply src/Controls/src/Core/FontImageSource.cs]: Fixed naming 

Right now nothing resizes when you change accessibility settings even the fonts.

I still need to look at WinUI for this one as well because the CanvasText doesn't have an obvious place to enable/disable scaling whereas iOS/Android seem to have obvious hooks

File: src/Core/src/Fonts/FontManager.iOS.cs [hunk 7]
```diff
@@ -41,7 +41,7 @@ public FontManager(IFontRegistrar fontRegistrar, ILogger<FontManager>? logger =
 		public UIFont GetFont(Font font, double defaultFontSize = 0) =>
 			GetFont(font, defaultFontSize, CreateFont);
 
-		public double GetFontSize(Font font, double defaultFontSize = 0) =>
+		double GetFontSize(Font font, double defaultFontSize = 0) =>
```
Comment (PureWeen) [inline src/Core/src/Fonts/FontManager.iOS.cs:44]: The only thing that was calling this currently was inside FontManager so it didn't seem worthwhile to expose this

File: src/Core/src/Fonts/IFontManager.iOS.cs [hunk 8]
```diff
@@ -8,7 +8,5 @@ public interface IFontManager
 		UIFont DefaultFont { get; }
 
 		UIFont GetFont(Font font, double defaultFontSize = 0);
-
-		double GetFontSize(Font font, double defaultFontSize = 0);
```
Comment (PureWeen) [inline src/Core/src/Fonts/IFontManager.iOS.cs:12]: Nothing is currently using this so I removed it

File: src/Core/src/Platform/Windows/TextBoxExtensions.cs [hunk 9]
```diff
@@ -78,14 +78,6 @@ public static void UpdatePlaceholderColor(this MauiTextBox textBox, IEditor edit
 		public static void UpdateFont(this MauiTextBox nativeControl, IText text, IFontManager fontManager) =>
 			nativeControl.UpdateFont(text.Font, fontManager);
 
-		public static void UpdateFont(this MauiTextBox nativeControl, Font font, IFontManager fontManager)
```
Comment (PureWeen) [inline src/Core/src/Platform/Windows/TextBoxExtensions.cs:81]: This is just duplicated code. There's already an extension method that takes "TextBox" with the same code

File: src/Core/src/Primitives/Font.cs [hunk 10]
```diff
@@ -3,61 +3,83 @@
 
 namespace Microsoft.Maui
 {
-	public struct Font
+	public readonly struct Font
```
Comment (PureWeen) [inline src/Core/src/Primitives/Font.cs:6]: we were already treating this as readonly so let's just make it official

File: src/Controls/src/Core/FontElement.cs (hunk 1 above)
Comment (PureWeen) [reply src/Controls/src/Core/FontElement.cs]: Cleaned up a little bit. There's still a bit more cleanup that could happen here but I'll leave that for a future PR

File: src/Core/src/Primitives/Font.cs (hunk 5 above)
Comment (PureWeen) [reply src/Core/src/Primitives/Font.cs]: I changed it all to AutoScalingEnabled

File: src/Controls/src/Core/Span.cs (hunk 2 above)
Comment (PureWeen) [reply src/Controls/src/Core/Span.cs]: All of our spans have a point where a UIFont or textsize is applied so it should work.

We currently don't have Spans implement on Maui though so we'll have to test once Spans are implemented

https://github.com/dotnet/maui/issues/1793

**Comment by arevellfraedom** [comment]
Great change 👍 
I can't quite tell what the defaults are for font scaling now on ios vs android? Because Android would have been on, iOS off before this change

**Comment by rachelkang** [comment]
@arevellfraedom font scaling will automatically be enabled across all the platforms including iOS now!

**Comment by arevellfraedom** [comment]
> @arevellfraedom font scaling will automatically be enabled across all the platforms including iOS now!

This is great! But also a little scary! (lots of app layouts will break until people fix them)
You _might_ want to have a way of turning this off globally to give people a chance to fix things up - it can be non-trivial to work out what to do.

**Comment by rachelkang** [comment]
@arevellfraedom yes definitely! so this PR does both of those things! It both:
1. Ensures font scaling will automatically be enabled across all the platforms including iOS
2. Introduces a `FontAutoScalingEnabled` boolean property that can be set to `false` if so desired

File: src/Controls/src/Core/DatePicker.cs [hunk 11]
```diff
@@ -29,6 +29,8 @@ public partial class DatePicker : View, IFontElement, ITextElement, IElementConf
 
 		public static readonly BindableProperty FontAttributesProperty = FontElement.FontAttributesProperty;
 
+		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnableProperty;
```
Comment (rachelkang) [inline src/Controls/src/Core/DatePicker.cs:32]: missing d
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnabledProperty;
```

File: src/Controls/src/Core/Editor.cs [hunk 12]
```diff
@@ -14,6 +14,8 @@ public partial class Editor : InputView, IEditorController, IFontElement, IEleme
 
 		public static readonly BindableProperty FontAttributesProperty = FontElement.FontAttributesProperty;
 
+		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnableProperty;
```
Comment (rachelkang) [inline src/Controls/src/Core/Editor.cs:17]: missing d
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnabledProperty;
```

File: src/Controls/src/Core/Entry.cs [hunk 13]
```diff
@@ -37,6 +37,8 @@ public partial class Entry : InputView, IFontElement, ITextAlignmentElement, IEn
 
 		public static readonly BindableProperty FontAttributesProperty = FontElement.FontAttributesProperty;
 
+		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnableProperty;
```
Comment (rachelkang) [inline src/Controls/src/Core/Entry.cs:40]: missing d
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnabledProperty;
```

File: src/Controls/src/Core/FontElement.cs [hunk 14]
```diff
@@ -24,6 +24,10 @@ static class FontElement
 		static readonly BindableProperty CancelEventsProperty =
 			BindableProperty.Create("CancelEvents", typeof(bool), typeof(FontElement), false);
 
+		public static readonly BindableProperty FontAutoScalingEnableProperty =
```
Comment (rachelkang) [inline src/Controls/src/Core/FontElement.cs:27]:
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty =
```

**Comment by arevellfraedom** [comment]
> @arevellfraedom yes definitely! so this PR does both of those things! It both:
> 
> 1. Ensures font scaling will automatically be enabled across all the platforms including iOS
> 2. Introduces a `FontAutoScalingEnabled` boolean property that can be set to `false` if so desired

That is per element though right? That would be slightly painful to go through and set false on everything while you work out how to make the design work. The impact is just more severe on iOS vs Android (which of course has been scaling forever) as AX5 is enormous, and requires some creative thinking to work out what to do (which in the end we just gave up on and limited to XXL).

File: src/Controls/src/Core/FontElement.cs [hunk 15]
```diff
@@ -43,53 +47,46 @@ static void OnFontPropertyChanged(BindableObject bindable, object oldValue, obje
 				bindable.ClearValue(FontFamilyProperty);
 				bindable.ClearValue(FontSizeProperty);
 				bindable.ClearValue(FontAttributesProperty);
+				bindable.ClearValue(FontAutoScalingEnableProperty);
```
Comment (rachelkang) [inline src/Controls/src/Core/FontElement.cs:50]:
```suggestion
				bindable.ClearValue(FontAutoScalingEnabledProperty);
```

File: src/Controls/src/Core/FontImageSource.cs [hunk 16]
```diff
@@ -42,5 +42,15 @@ public double Size
 			get => (double)GetValue(SizeProperty);
 			set => SetValue(SizeProperty, value);
 		}
+
+		public static readonly BindableProperty FontAutoScalingEnableProperty =
```
Comment (rachelkang) [inline src/Controls/src/Core/FontImageSource.cs:46]:
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty =
```

File: src/Controls/src/Core/FontImageSource.cs [hunk 17]
```diff
@@ -42,5 +42,15 @@ public double Size
...
+			BindableProperty.Create("FontAutoScalingEnable", typeof(bool), typeof(FontImageSource), false,
+				propertyChanged: (b, o, n) => ((FontImageSource)b).OnSourceChanged());
+
+		public bool FontAutoScalingEnable
+		{
+			get => (bool)GetValue(FontAutoScalingEnableProperty);
```
Comment (rachelkang) [inline src/Controls/src/Core/FontImageSource.cs:52]:
```suggestion
			get => (bool)GetValue(FontAutoScalingEnabledProperty);
```

File: src/Controls/src/Core/FontImageSource.cs [hunk 18]
```diff
@@ -42,5 +42,15 @@ public double Size
...
+			set => SetValue(FontAutoScalingEnableProperty, value);
```
Comment (rachelkang) [inline src/Controls/src/Core/FontImageSource.cs:53]:
```suggestion
			set => SetValue(FontAutoScalingEnabledProperty, value);
```

File: src/Controls/src/Core/Label.cs [hunk 19]
```diff
@@ -28,6 +28,8 @@ public partial class Label : View, IFontElement, ITextElement, ITextAlignmentEle
 
 		public static readonly BindableProperty FontAttributesProperty = FontElement.FontAttributesProperty;
 
+		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnableProperty;
```
Comment (rachelkang) [inline src/Controls/src/Core/Label.cs:31]:
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnabledProperty;
```

File: src/Controls/src/Core/Picker.cs [hunk 20]
```diff
@@ -40,6 +40,8 @@ public partial class Picker : View, IFontElement, ITextElement, ITextAlignmentEl
 
 		public static readonly BindableProperty FontAttributesProperty = FontElement.FontAttributesProperty;
 
+		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnableProperty;
```
Comment (rachelkang) [inline src/Controls/src/Core/Picker.cs:43]:
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnabledProperty;
```

File: src/Controls/src/Core/RadioButton.cs [hunk 21]
```diff
@@ -59,6 +59,8 @@ public partial class RadioButton : TemplatedView, IElementConfiguration<RadioBut
 
 		public static readonly BindableProperty FontSizeProperty = FontElement.FontSizeProperty;
 
+		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnableProperty;
```
Comment (rachelkang) [inline src/Controls/src/Core/RadioButton.cs:62]:
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnabledProperty;
```

File: src/Controls/src/Core/SearchBar.cs [hunk 22]
```diff
@@ -26,6 +26,8 @@ public partial class SearchBar : InputView, IFontElement, ITextAlignmentElement,
 
 		public static readonly BindableProperty FontAttributesProperty = FontElement.FontAttributesProperty;
 
+		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnableProperty;
```
Comment (rachelkang) [inline src/Controls/src/Core/SearchBar.cs:29]:
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnabledProperty;
```

File: src/Controls/src/Core/Shell/SearchHandler.cs [hunk 23]
```diff
@@ -132,6 +132,8 @@ public Color TextColor
 
 		public static readonly BindableProperty FontAttributesProperty = FontElement.FontAttributesProperty;
 
+		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnableProperty;
```
Comment (rachelkang) [inline src/Controls/src/Core/Shell/SearchHandler.cs:135]:
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnabledProperty;
```

File: src/Controls/src/Core/Span.cs [hunk 24]
```diff
@@ -76,6 +76,8 @@ public string Text
 
 		public static readonly BindableProperty FontAttributesProperty = FontElement.FontAttributesProperty;
 
+		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnableProperty;
```
Comment (rachelkang) [inline src/Controls/src/Core/Span.cs:79]:
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnabledProperty;
```

File: src/Controls/src/Core/TimePicker.cs [hunk 25]
```diff
@@ -24,6 +24,8 @@ public partial class TimePicker : View, IFontElement, ITextElement, IElementConf
 
 		public static readonly BindableProperty FontAttributesProperty = FontElement.FontAttributesProperty;
 
+		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnableProperty;
```
Comment (rachelkang) [inline src/Controls/src/Core/TimePicker.cs:27]:
```suggestion
		public static readonly BindableProperty FontAutoScalingEnabledProperty = FontElement.FontAutoScalingEnabledProperty;
```

**Comment by PureWeen** [comment]
@arevellfraedom  I have no research data to back this up :-) But I feel like people who are font scaling conscious would have already accounted for this in their app and the changes they make will be removing hacks they had to add into iOS in order to accommodate font scaling. If they are developing an app that's on iOS/Android/WinUI the fonts already scaled on WinUI/Android so they already would have had to make the layouts work. MAUI also does a better job in general with layouts and scaled fonts. So, I do think this will be a breaking change for people but I feel like in general what scaling conscious users will end up with is a more symmetrical app across iOS/Android/WinUI instead of having to make special accommodations on iOS

**Comment by PureWeen** [comment]
@arevellfraedom on a side note :-) all of the font code is handled by `IFontManager` so if someone wanted to disable autoscaling at a global level they could register their own `IFontManager` and then modify how we process the FontAutoScalingEnabled properties

File: src/Controls/src/Core/Platform/Android/Shell/ShellToolbarTracker.cs [hunk 26]
```diff
@@ -675,11 +676,19 @@ protected override void Dispose(bool disposing)
...
-				_defaultSize = fontManager.GetFontSize(Font.OfSize("Roboto", 0));
+				TintColor = defaultColor;		
+				if (context.TryResolveAttribute(AndroidResource.Attribute.TextSize, out float? value) &&
+					value != null)
+				{
+					_defaultSize = value.Value;
+				}
+				else
+				{
+					_defaultSize = 50;
```
Comment (rachelkang) [inline src/Controls/src/Core/Platform/Android/Shell/ShellToolbarTracker.cs:689]: Where did this 50 come from?

File: src/Controls/src/Core/TimePicker.cs [hunk 27]
```diff
@@ -86,19 +94,29 @@ public virtual string UpdateFormsText(string source, TextTransform textTransform
...
 
 		void IFontElement.OnFontAttributesChanged(FontAttributes oldValue, FontAttributes newValue) =>
+			HandleFontChanged();
+
+		void IFontElement.OnFontAutoScalingEnabledChanged(bool oldValue, bool newValue) =>
+			HandleFontChanged();
+
+		void HandleFontChanged()
+		{
+			// Null out the Maui font value so it will be recreated next time it's accessed
```
Comment (rachelkang) [inline src/Controls/src/Core/TimePicker.cs:116]: Should this comment be added everywhere the code is the same in other controls?

File: src/Core/src/Fonts/FontManager.iOS.cs [hunk 28]
```diff
@@ -155,10 +155,19 @@ UIFont CreateFont(Font font)
 			{
 				var defaultFont = UIFont.SystemFontOfSize(size);
 				var descriptor = defaultFont.FontDescriptor.CreateWithAttributes(GetFontAttributes(font));
-				return UIFont.FromDescriptor(descriptor, size);
+				return ApplyScaling(font, UIFont.FromDescriptor(descriptor, size));
 			}
 
-			return UIFont.SystemFontOfSize(size);
+			return ApplyScaling(font, UIFont.SystemFontOfSize(size));
+
```
Comment (rachelkang) [inline src/Core/src/Fonts/FontManager.iOS.cs:162]: nit: remove extra line

File: src/Core/src/Platform/Android/ContextExtensions.cs [hunk 29]
```diff
@@ -121,6 +121,9 @@ internal static int GetDisabledThemeAttrColor(this Context context, int attr)
 			}
 		}
 
+		public static bool TryResolveAttribute(this Context context, int id, out float? value) =>
```
Comment (rachelkang) [inline src/Core/src/Platform/Android/ContextExtensions.cs:124]: Why is this needed?

File: src/Controls/src/Core/TimePicker.cs (hunk 27 above)
Comment (PureWeen) [reply src/Controls/src/Core/TimePicker.cs:116]: This is fine for now. This all needs to be refactored out anyway but I kind of hit my limit of refactoring for this PR

**Comment by arevellfraedom** [comment]
> @arevellfraedom I have no research data to back this up :-) But I feel like people who are font scaling conscious would have already accounted for this in their app and the changes they make will be removing hacks they had to add into iOS in order to accommodate font scaling. If they are developing an app that's on iOS/Android/WinUI the fonts already scaled on WinUI/Android so they already would have had to make the layouts work. MAUI also does a better job in general with layouts and scaled fonts. So, I do think this will be a breaking change for people but I feel like in general what scaling conscious users will end up with is a more symmetrical app across iOS/Android/WinUI instead of having to make special accommodations on iOS

Also no research, just personal experience! Your comments are correct, but there is just one aspect that got us and forced us to put more thought in than we experienced - which is that iOS sizes scale MUCH higher than Android. 

So generally, the layouts we had done worked mostly OK in Android once we tried out turning up the size (which generally scales by approx 130%), except for the odd thing to correct. But iOS was a whole different story. Up in the AX range it is getting close to 200%, which made a large proportion of the layout just explode. So that is what I am concerned with - devs being suddenly forced to deal with a much more significant problem on iOS than they ever would have seen on Android. But as long as their is a way to get them out of it while they figure out a way, that works ok. 

As I say we ended up capping iOS to a certain size (XXL) as anything above that was just impossible.

File: src/Core/src/Platform/Android/ContextExtensions.cs (hunk 29 above)
Comment (PureWeen) [reply src/Core/src/Platform/Android/ContextExtensions.cs:124]: for this
https://github.com/dotnet/maui/pull/1774/files/1583bbb10df6a0b72581d12b008fbab4864b34e3#diff-ce2735a0580c21a5e576b81a84822093bf8bee64175e0d12700c5c50dc99ed44R682

File: src/Controls/src/Core/Platform/Android/Shell/ShellToolbarTracker.cs (hunk 26 above)
Comment (PureWeen) [reply src/Controls/src/Core/Platform/Android/Shell/ShellToolbarTracker.cs:689]: Wherever this came from :-) 

https://github.com/dotnet/maui/blob/c6efff91de8fe30fadbcdfff24f7b9af96dc0c91/src/Compatibility/Core/src/Android/Forms.cs#L159

It most likely won't ever get set to 50 since the only purpose was for pre-lollipop which we don't support.

I just left here for backwards compatibility

**Comment by mattleibow** [comment]
For disabling scaling across the board (and I didn't type this message) you can use an implicit style on the views to disable them. But this is probably worse than a dodgy UI. Rather have an ugly app that a partially bind person can use instead of a beautiful app that my dad can't even use with his "slightly going" vision.

It all depends on the app of course.

**Comment by arevellfraedom** [comment]
> For disabling scaling across the board (and I didn't type this message) you can use an implicit style on the views to disable them. But this is probably worse than a dodgy UI. Rather have an ugly app that a partially bind person can use instead of a beautiful app that my dad can't even use with his "slightly going" vision.
> 
> It all depends on the app of course.

Unfortunately a broken app due to not handling enormous iOS sizes is often neither beautiful to the fully sighted nor usable to the partially. Truely broken layouts most often manifest with clipped/truncated text, which literally can't be read.

I'm not against it being on by default, that will probably result in better accessibility for most people. But some very clear guidance might need to be given in release notes etc that it should be tested in iOS now, and if things are really bad what a plan of attack may be for the developer mitigate it until they get a chance to correct it (either element-by-element, implicit style or handler override).

**Review by Redth — APPROVED** [review]

**Comment by PureWeen** [comment]
> > For disabling scaling across the board (and I didn't type this message) you can use an implicit style on the views to disable them. But this is probably worse than a dodgy UI. Rather have an ugly app that a partially bind person can use instead of a beautiful app that my dad can't even use with his "slightly going" vision.
> > It all depends on the app of course.
> 
> Unfortunately a broken app due to not handling enormous iOS sizes is often neither beautiful to the fully sighted nor usable to the partially. Truely broken layouts most often manifest with clipped/truncated text, which literally can't be read.
> 
> I'm not against it being on by default, that will probably result in better accessibility for most people. But some very clear guidance might need to be given in release notes etc that it should be tested in iOS now, and if things are really bad what a plan of attack may be for the developer mitigate it until they get a chance to correct it (either element-by-element, implicit style or handler override).

Agreed!! We'll make sure to have some clarifying docs so users will know how they can disable it globally and include it inside our breaking changes

**Comment by beeradmoore** [comment]
> 
> For Android/WinUI the default behavior here doesn't change.
> 
> This now means that fonts on iOS will auto scale based on the dynamic font sizing settings on the device
> 
> ...
> 
> ### Additions made
> * Adds Font.AutoScalingEnabled
> * Adds Button.FontAutoScalingEnabled
> * Adds DatePicker.FontAutoScalingEnabled
> * Adds Editor.FontAutoScalingEnabled
> * Adds Entry.FontAutoScalingEnabled
> * Adds FontElement.FontAutoScalingEnabled
> * Adds Label.FontAutoScalingEnabled
> * Adds Picker.FontAutoScalingEnabled
> * Adds RadioButton.FontAutoScalingEnabled
> * Adds SearchBar.FontAutoScalingEnabled
> * Adds Span.FontAutoScalingEnabled
> * Adds TimePicker.FontAutoScalingEnabled
> * Adds FontImageSource.FontAutoScalingEnabled

Just to clarify, setting `FontAutoScalingEnabled = false` on Android does have the functionality in maui to precent fonts adjusting? (by default they always had adjusted in Xamarin.Forms)

**Comment by heybuddiescode** [comment]
Hej - I'm a bit late to the game, however I'm really missing a way to turn this off globally. The FontAutoScalingEnabled for some reason is not respected when applied as a Xaml Styling, therefore breaking our whole app.

Eg. this style in a separate file, every other attribute is respected, except for the FontAutoScalingEnabled.
<Style x:Key="TextSmall" TargetType="Label">
        <Setter Property="TextColor" Value="{StaticResource PrimaryColor}" />
        <Setter Property="FontFamily" Value="SourceSansPro-Regular" />
        <Setter Property="FontSize" Value="Small" />
        <Setter Property="FontAutoScalingEnabled" Value="False"/>
    </Style>

**Comment by mattleibow** [comment]
@heybuddiescode can you open a new issue for this?
